src/mame/drivers/ymdx7.cpp

```diff
@@ -0,0 +1,417 @@
+// license:BSD-3-Clause
+// copyright-holders:David Viens
+/***************************************************************************
+
+    Driver for Yamaha DX7
+    Copyright 2011-2021 David Viens
+
+    Main CPU is a HD63B03RP
+    Sub  CPU is a HD6805S1P-A33 (ROM on chip) 
+
+    The FM sound system is done by two discreet chips:
+    OPS(YM2128) and EGS(YM2129)
+
+    ///////////////// ////////////////////////////////////////////////////////
+    Main CPU PCB fixed pins:
+
+    PIN02:  XTAL : NC
+    PIN03:  EXTAL: 4.713250MHz (internally divided by 4 to get 1.178312MHz)
+    PIN04: !NMI  : 1 (4.7K pull-up)
+    PIN05: !IRQ  : 1 (4.7K pull-up) but... linked to Sub CPU
+
+    PIN07: !STBY : 1 (4.7K pull-up)
+    /////////////////////////////////////////////////////////////////////////
+    RAM access is done through logic on the PCB when
+    bit 6 of port 5 (RAME) is low.
+
+    PORT1:direction 0 (all bits read only!)
+    P10 to P17 : 8bits that are mapped to Sub CPU's shared Address/Data line
+
+    PORT2 direction 0x11 (XXX10001)
+    P20 : write to   TC4053's X (pin14) (SUB CPU BUSY)
+    P21 : read  from TC4053's Y (pin15) (SUB CPU handshake)
+    P22 : read  from TC4053's Z (pin 4) (250KHz Clock)
+    P23 : read  from MIDI IN
+    P24 : write to   MIDI OUT
+****************************************************************************/
+
+#include "emu.h"
+//#include "bus/midi/midi.h"
+#include "cpu/m6800/m6801.h"
+#include "cpu/m6805/m68705.h"
+#include "machine/adc0808.h"
+#include "machine/nvram.h"
+#include "video/hd44780.h"
+#include "emupal.h"
+#include "screen.h"
+//#include "speaker.h"
+
+constexpr auto DX7CLOCK    = 9'426'500;
+constexpr auto NUM_ANALOG  = 7;
+constexpr auto NUM_INITIAL = (2*NUM_ANALOG);
+
+#define TRACE printf
```

Comment:
It would be better to use `logmacro.h` here. It can be configured to either output to a log or the standard output.